In [1]:
import pandas as pd
from mlxtend.preprocessing import TransactionEncoder
from mlxtend.frequent_patterns import association_rules
from mlxtend.frequent_patterns import apriori

In [2]:
ds_asso = [["소주",'콜라', '맥주'],
          ['소주','콜라','와인'],
          ['소주', '주스',],
          ['콜라','맥주',],
           ['소주','콜라','맥주','와인',],
           ['주스']]

In [3]:
enc = TransactionEncoder()
arr = enc.fit_transform(X=ds_asso)
ds_pre = pd.DataFrame(arr, columns= enc.columns_)
ds_pre

맥주     소주     와인     주스     콜라
0   True   True  False  False   True
1  False   True   True  False   True
2  False   True  False   True  False
3   True  False  False  False   True
4   True   True   True  False   True
5  False  False  False   True  False

In [4]:
min_support = 0.5
ds_freq = apriori(ds_pre, min_support = 0.5, use_colnames = True)

ds_freq.round(3)

support  itemsets
0    0.500      (맥주)
1    0.667      (소주)
2    0.667      (콜라)
3    0.500  (콜라, 맥주)
4    0.500  (소주, 콜라)

In [5]:
# 기준 = 신뢰도, 최소 신뢰도 = 0.5
ds_asso_rule = association_rules(ds_freq, metric = "confidence", min_threshold=0.5)
ds_asso_rule.round(3)

antecedents consequents  antecedent support  consequent support  support  \
0        (콜라)        (맥주)               0.667               0.500      0.5   
1        (맥주)        (콜라)               0.500               0.667      0.5   
2        (소주)        (콜라)               0.667               0.667      0.5   
3        (콜라)        (소주)               0.667               0.667      0.5   

   confidence   lift  leverage  conviction  
0        0.75  1.500     0.167       2.000  
1        1.00  1.500     0.167         inf  
2        0.75  1.125     0.056       1.333  
3        0.75  1.125     0.056       1.333

* [(맥주), (콜라)] Confidence = 1.0 : 맥주를 구매한 고객은 콜라도 반드시 구매
* [(콜라), (맥주)] Confidence = 0.75 : 콜라를 구매한 고객 중 맥주를 고른 고객은 0.75
* [(콜라), (소주)] Lift = 1.125 [(콜라), (맥주)] Lift = 1.5 : 콜라를 구매한 고객들에게는 소주보다 맥주를 추천

In [7]:
# 기준 = 신뢰도, 최소 신뢰도 = 0.5
ds_asso_rule = association_rules(ds_freq, metric='confidence', min_threshold=0.5)
ds_asso_rule.round(3)

antecedents consequents  antecedent support  consequent support  support  \
0        (콜라)        (맥주)               0.667               0.500      0.5   
1        (맥주)        (콜라)               0.500               0.667      0.5   
2        (소주)        (콜라)               0.667               0.667      0.5   
3        (콜라)        (소주)               0.667               0.667      0.5   

   confidence   lift  leverage  conviction  
0        0.75  1.500     0.167       2.000  
1        1.00  1.500     0.167         inf  
2        0.75  1.125     0.056       1.333  
3        0.75  1.125     0.056       1.333